routes/Program/controllers.js

```diff
@@ -9,24 +9,140 @@ const router = express.Router();
+      try {
+         res.status(200).json(await service.getByCollegeAndDegree(req.query.collegeId, req.query.degree, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree == null && req.query.language != null && req.query.type == null && req.query.collegeId != null) {
+      try {
+         res.status(200).json(await service.getByCollegeAndLanguage(req.query.collegeId, req.query.language, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree == null && req.query.language == null && req.query.type != null && req.query.collegeId != null) {
+      try {
+         res.status(200).json(await service.getByCollegeAndType(req.query.collegeId, req.query.type, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree != null && req.query.language != null && req.query.type == null && req.query.collegeId == null) {
+      try {
+         res.status(200).json(await service.getByDegreeAndLanguage(req.query.degree, req.query.language, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree != null && req.query.language == null && req.query.type != null && req.query.collegeId == null) {
+      try {
+         res.status(200).json(await service.getByDegreeAndType(req.query.degree, req.query.type, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree == null && req.query.language != null && req.query.type != null && req.query.collegeId == null) {
+      try {
+         res.status(200).json(await service.getByLanguageAndType(req.query.language, req.query.type, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree != null && req.query.language != null && req.query.type == null && req.query.collegeId != null) {
+      try {
+         res.status(200).json(await service.getByCollegeAndDegreeAndLanguage(req.query.collegeId, req.query.degree, req.query.language, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree != null && req.query.language == null && req.query.type != null && req.query.collegeId != null) {
+      try {
+         res.status(200).json(await service.getByCollegeAndDegreeAndType(req.query.collegeId, req.query.degree, req.query.type, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else if (req.query.degree != null && req.query.language != null && req.query.type != null && req.query.collegeId != null) {
+      try {
+         res.status(200).json(await service.getByCollegeAndDegreeAndLanguageAndType(req.query.collegeId,
+             req.query.degree, req.query.language, req.query.type, querySchema));
+      } catch (err) {
+         next(err);
+      }
+   } else { //empty query params => getAll
+      try {
+         res.status(200).json(await service.getAll(req.query));
```

Comment:
`getAll` could be implemented with `getFiltered` as well, because it's paginated, but if no conditions are passed to `getFiltered`, we get everything so a `getAll` in essence.

Reply:
done